### PR TITLE
feat: do not run headless before snapshot finished

### PR DESCRIPTION
**Pull request by moreal**
It makes the launcher not start the headless before snapshot downloading ended up.

- [x] Use `chain tip` subcommand to check it should download snapshot or not.
- [x] Use `key *` subcommand to fetch and handle key store.
- [x] Use `validation private-key` subcommand to validate private-key.
- [x] Use `key convert --address [PRIVATE-KEY]` to convert private key into address format.

See also https://github.com/planetarium/NineChronicles.Headless/issues/342, https://github.com/planetarium/NineChronicles.Headless/issues/355, https://github.com/planetarium/NineChronicles.Headless/pull/367

### CODE REVIEW

**Comment by moreal** [comment]
I opened this pull reqeust though the build failed, because it fails by `Lib9c` hash difference between `NineChronicles` and `NineChronicles.Headless`.

**Comment by ipdae** [comment]
check-submodule is failed.

File: src/renderer/views/login/MiningView.tsx [hunk 1]
```diff
@@ -15,6 +15,7 @@ const MiningView = observer(
     const classes = miningViewStyle();
     const setMining = (isMining: boolean) => {
       routerStore.push("/lobby/preload");
+      console.log("set-private-key setMining", accountStore.privateKey);
```
Comment (longfin) [inline src/renderer/views/login/MiningView.tsx:18]: I guess it should be removed ㊙️

**Review by limebell — APPROVED** [review]
👍

**Review by area363 — APPROVED** [review]

File: src/renderer/views/login/MiningView.tsx (hunk 1 above)
Comment (moreal) [reply src/renderer/views/login/MiningView.tsx]: I removed it in 1bde93b

**Pull request merged**